apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.BenchmarkingTool/Program.cs

```diff
@@ -0,0 +1,244 @@
+                Read(configuration, client, logger, "Read[1]");
+                Read(configuration, client, logger, "Read[2]");
+                Read(configuration, client, logger, "Read[3]");
+            }
+            finally
+            {
+                DeleteBucket(client, configuration.Bucket);
+            }
+        }
+
+        private static Bucket CreateBucket(string bucketName, StorageClient client)
+        {
+            var bucket = client.CreateBucket(s_projectId, new Bucket { Name = bucketName });
+            SleepAfterBucketCreateDelete();
+            return bucket;
+        }
+
+        private static void DeleteBucket(StorageClient client, string bucketName)
+        {
+            try
+            {
+                client.DeleteBucket(bucketName, new DeleteBucketOptions { UserProject = null, DeleteObjects = true });
+            }
+            catch (GoogleApiException)
+            {
+                //Some tests fail to delete buckets due to object retention locks etc.
+                //They can be cleaned up later.
+            }
+            SleepAfterBucketCreateDelete();
+        }
+
+        private static void CreateRandomFile(Configuration configuration)
+        {
+            var random = new Random();
+            long bytesLeft = configuration.DataSizeInBytes;
+            byte[] buffer = new byte[1024 * 1024];
+            using var output = File.Create(configuration.LocalFile);
+            while (bytesLeft > 0)
+            {
+                random.NextBytes(buffer);
+                int count = (int) Math.Min(bytesLeft, buffer.Length);
+                output.Write(buffer, 0, count);
+                bytesLeft -= count;
+            }
+        }
+
+        private static void Write(Configuration configuration, StorageClient client, PerformanceLogger logger, string opName)
+        {
+            string elapsedTimeUs = "-";
+            bool isSucccess = false;
+
+            var logs = new string[10]; //logs array should be in same order as s_logHeaders.
+            try
+            {
+                CreateRandomFile(configuration);
+
+                UploadObjectOptions options = null;
+
+                if (configuration.UploadChunkSizeInBytes != 0)
+                {
+                    options = new UploadObjectOptions { ChunkSize = configuration.UploadChunkSizeInBytes };
+                }
+
+                long totalElapsedMicroseconds = 0;
+
+
+                logs[0] = opName;
+                logs[1] = configuration.DataSizeInBytes.ToString(CultureInfo.InvariantCulture);
+                logs[2] = "NA";
+                logs[3] = options != null ? options.ChunkSize.ToString() : ResumableUpload.DefaultChunkSize.ToString();
+                logs[4] = "TRUE";
+                logs[5] = "NA";
+                logs[6] = "JSON";
+                logs[7] = "-1";
+
+                Console.WriteLine("Uploading..");
+                using (var input = File.OpenRead(configuration.LocalFile))
```

Comment:
No need for a block here - just `using var input = File.OpenRead(configuration.LocalFile);`